### PR TITLE
PRM-549 Send audit event on successful appeal

### PR DESCRIPTION
**Pull request by ruthdas03**

### CODE REVIEW

**Comment by ruthdas03** [comment]
Sample audit event as generated in local (message from DATASTREAM using Wireshark)
{
   "auditSource":"penalties-appeals-frontend",
   "auditType":"PenaltyAppealSubmitted",
   "eventId":"4020bf14-051a-480a-be36-6e8d010e721b",
   "tags":{
      "clientIP":"-",
      "path":"/penalties-appeals/check-your-answers",
      "X-Session-ID":"session-db753aba-7f80-458b-a709-a3561335d8c7",
      "Akamai-Reputation":"-",
      "X-Request-ID":"govuk-tax-ab727da7-92d7-48b7-b34c-76af488d523d",
      "deviceID":"mdtpdi#d8bb3fee-d73f-4202-bdd6-0e541cc09943#1624452860754_/sEwmzPRXAT11afJRWThrw==",
      "clientPort":"-",
      "transactionName":"penalties-appeal-submitted"
   },
   "detail":{
      "submittedBy":"client",
      "taxIdentifier":"234060050",
      "identifierType":"VRN",
      "penaltyId":"1234567891",
      "penaltyType":"LSP",
      "appealInformation":{
         "type":"fireOrFlood",
         "startDateOfEvent":"2021-02-01",
         "lateAppeal":false
      }
   },
   "generatedAt":"2021-09-02T08:26:31.379+0000",
   "metadata":{
      "sendAttemptAt":"2021-09-02T08:26:31.380+0000",
      "instanceID":"2ba5add2-adf8-48ea-bea0-23dde6008263",
      "sequence":24
   }
}

File: test/services/AppealServiceSpec.scala [hunk 1]
```diff
@@ -226,6 +228,11 @@ class AppealServiceSpec extends SpecBase {
         result shouldBe true
       }
 
+      "call the audit method in audit service after appeal is submitted successfully (in the above submitAppeal)" in new Setup{
+          verify(mockAuditService, times(1)).audit(ArgumentMatchers.any[JsonAuditModel])(ArgumentMatchers.any[HeaderCarrier],
```
Comment (LewisMitchell) [inline test/services/AppealServiceSpec.scala:232]: Could you not add this to the above test? And just extend the test description to be something like:
`parse the session keys into a model and return true when the connector call is successful and audit the response` as it seems a bit fragile to put this test in without doing any calls.

**Review by LewisMitchell — CHANGES_REQUESTED** [review]
Looks good - AT's fine, just a query about a test

File: test/services/AppealServiceSpec.scala (hunk 1 above)
Comment (ruthdas03) [reply test/services/AppealServiceSpec.scala:232]: audit() returns 'Unit' not sure how the result against the audit call can be validated with true.

Comment (LewisMitchell) [reply test/services/AppealServiceSpec.scala:232]: So if you remove this test, move the verify call to the test above that. It's all done in one test?

Comment (ruthdas03) [reply test/services/AppealServiceSpec.scala]: Yes sure, Done (Apologies, misinterpreted your initial review!)

**Pull request merged**